bytecomp/matching.ml

```diff
@@ -2687,7 +2687,7 @@ let rec name_pattern default = function
 let arg_to_var arg cls = match arg with
 | Lvar v -> v,arg
 | _ ->
-    let v = name_pattern "match" cls in
+    let v = name_pattern "*match*" cls in
```

Comment:
This is a hack to hide this kind of variable from being seen in the debugger.  Ideally we would have a flag on `Ident.t` to enable this to be represented.  I did wonder about removing this change from this patch but in fact I think it should maybe stay, and we should consider the `Ident.t` change in due course (which could clean up "*opt", etc.)